src/js/providers/providers.js

```diff
@@ -40,11 +40,14 @@ Object.assign(Providers.prototype, {
             let loadProvider = false;
             for (let i = playlist.length; i--;) {
                 const item = playlist[i];
-                const supported = this.providerSupports(provider, item.sources[0]);
-                if (supported) {
-                    playlist.splice(i, 1);
+                const source = item.sources[0];
+                if (source) {
+                    const supported = this.providerSupports(provider, item.sources[0]);
```

Comment:
`item.sources[0]` is called twice. I don't see why this `source` check is necessary.